app/assets/javascripts/comments.js

```diff
@@ -30,14 +30,14 @@ function init_comments_page() {
     });
     // end of removing comments
 
-
-    // comments autoload
-    var comment_page = 1;
-    load_comments();
-    // end of comments autoload
+    init_infinity_scroll(1, comments_source_url());
 
     // Helpers
 
+    function comments_source_url() {
+        return window.location.href + '/comments'
```

Comment:
for routes generation try to use this gem https://github.com/railsware/js-routes